CMakeLists.txt

```diff
@@ -2641,15 +2641,35 @@ endif()
 # build netdatacli
 #
 
+include(FetchContent)
+include(NetdataFetchContentExtra)
+
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.28)
+        FetchContent_Declare(
+                CLI11
+                GIT_REPOSITORY https://github.com/CLIUtils/CLI11.git
+                GIT_TAG 6c7b07a878ad834957b98d0f9ce1dbe0cb204fc9 # v2.4.2
+                EXCLUDE_FROM_ALL
+        )
+else()
+        FetchContent_Declare(CLI11
+                GIT_REPOSITORY https://github.com/microsoft/mimalloc.git
+                GIT_REPOSITORY https://github.com/CLIUtils/CLI11.git
+                GIT_TAG 6c7b07a878ad834957b98d0f9ce1dbe0cb204fc9 # v2.4.2
+        )
+endif()
+
+FetchContent_MakeAvailable_NoInstall(CLI11)
```

Comment:
I thought our goal was to minimize the amount of stuff like this in the main CMakeLists.txt file?